src/util.ts

```diff
@@ -38,12 +39,77 @@ export function nop() {
 /**
  * Find the tsconfig.json, read it, and return parsed contents.
  * @param rootDir Directory where the tsconfig.json should be found.
+ * If the tsconfig.json file has an "extends" field hop down the dependency tree until it ends
  */
+
+
+export interface ConfigFile {
+  files?: string[];
+  compilerOptions?: {};
+  include?: string[];
+  exclude?: string[];
+}
+
 export async function getTSConfig(
-    rootDir: string, customReadFilep?: ReadFileP): Promise<{}> {
+    rootDir: string, customReadFilep?: ReadFileP): Promise<ConfigFile> {
+  // array to hold accessed files, used to check for circular references
+
+
+
   const tsconfigPath = path.join(rootDir, 'tsconfig.json');
+
   customReadFilep = customReadFilep || readFilep;
   const json = await customReadFilep(tsconfigPath, 'utf8');
-  const contents = JSON.parse(json);
-  return contents;
+  let contents = JSON.parse(json);
+
+  const readArr = ['tsconfig.json'];
+  if (contents['extends']) {
+    const extension =
+        await getExtension(contents['extends'], customReadFilep, readArr);
+    contents = combineTSConfig(extension, contents);
+  }
+  return Promise.resolve(contents);
 }
+
+async function getExtension(
+    filePath: string, customReadFilep: ReadFileP,
+    readFiles: string[]): Promise<ConfigFile> {
+  customReadFilep = customReadFilep || readFilep;
+  if (!filePath) {
+    throw new Error('Undefined passed in');
+  }
+  if (readFiles.indexOf(filePath) !== -1) {
+    // did I throw the error correctly?
+    throw new Error('Circular Reference Detected');
+  }
+  readFiles.push(filePath);
```

Comment:
`filePath` may be relative to the location of `tsconfig.json`. This might result in issues if say, `tsconfig.json` depends on `foo/tsconfig.bar.json` which depends on `tsconfig.json`: does that last file refer to `./tsconfig.json` or `./foo/tsconfig.json`?

Reply:
👍